sdk-extensions/aws/src/main/java/io/opentelemetry/sdk/extension/aws/trace/SamplingRuleApplier.java

```diff
@@ -40,7 +43,11 @@
     XRAY_CLOUD_PLATFORM = Collections.unmodifiableMap(xrayCloudPlatform);
   }
 
+  private final String clientId;
+  private final String ruleName;
```

Comment:
As far as I can tell ruleName is used only for statistics purposes. Therefore could be moved to that class (set in constructor). Then, in order to reduce some boilerplate builder code SamplingStatisticsDocument.Builder could be manufactured out of Statistics class.

Reply:
Just a nice to have & consistent with the current X-Ray SDK behavior AFAIK. Not blocking on this PR.